components/sup/src/manager/service/config.rs

```diff
@@ -115,14 +116,22 @@ pub struct Cfg {
 
     /// Source of the user configuration
     pub user_config_path: UserConfigPath,
+
+    /// The path to an optional dev-time configuration directory that
+    /// is being used.
+    override_config_dir: Option<PathBuf>
 }
 
 impl Cfg {
     pub fn new<P: PackageConfigPaths>(package: &P, config_from: Option<&PathBuf>) -> Result<Cfg> {
-        let pkg_root = config_from.and_then(|p| Some(p.clone())).unwrap_or(
-            package.default_config_dir(),
-        );
-        let default = Self::load_default(pkg_root)?;
+        let override_config_dir = config_from.and_then(|c| Some(c.clone()));
+        let default = {
+            let pkg_root = match override_config_dir {
+                Some(ref path) => Cow::Borrowed(path),
+                None => Cow::Owned(package.default_config_dir())
+            };
```

Comment:
Getting clever with the `Cow`s! In the first case we have a borrowed `Path` and the second an owned `PathBuf` so rather than cloning the first case or assigning the second case to a variable (so it can be borrowed) you get the benefit of your answer without resorting to either. Nicely done 😉